jabgui/src/main/java/org/jabref/gui/entryeditor/citationrelationtab/CitationRelationsTab.java

```diff
@@ -434,13 +444,29 @@ protected void bindToEntry(BibEntry entry) {
      */
     private void searchForRelations(BibEntry entry, CheckListView<CitationRelationItem> listView, Button abortButton,
                                     Button refreshButton, CitationFetcher.SearchType searchType, Button importButton,
-                                    ProgressIndicator progress) {
+                                    ProgressIndicator progress, boolean shouldRefresh) {
         if (entry.getDOI().isEmpty()) {
             hideNodes(abortButton, progress);
             showNodes(refreshButton);
             listView.getItems().clear();
-            listView.setPlaceholder(
-                    new Label(Localization.lang("The selected entry doesn't have a DOI linked to it. Lookup a DOI and try again.")));
+            Text doiLookUpText = new Text(Localization.lang("The selected entry doesn't have a DOI linked to it."));
+            Hyperlink doiLookUpHyperLink = new Hyperlink(Localization.lang("Lookup a DOI and try again."));
+            TextFlow doiLookUpTextFlow = new TextFlow(doiLookUpText, doiLookUpHyperLink);
+            Label placeHolder = new Label("", doiLookUpTextFlow);
+            doiLookUpHyperLink.setOnAction(e -> {
+                CrossRef doiFetcher = new CrossRef();
+                BackgroundTask.wrap(() -> doiFetcher.findIdentifier(entry))
+                              .onRunning(() -> listView.setPlaceholder(new Label("Looking up DOI...")))
+                              .onSuccess(doiIdentifier -> {
+                                  if (doiIdentifier.isPresent()) {
+                                      entry.setField(StandardField.DOI, doiIdentifier.get().asString());
+                                      searchForRelations(entry, listView, abortButton, refreshButton, searchType, importButton, progress, shouldRefresh);
```

Comment:
you are introducing a recursive cascade here with endless creation of UI elements.
This needs to be done differently.

Extract the background task. put the result in a property. update the view if the property is modified.